packages/api/src/resolvers/type/media-types.ts

```diff
@@ -1,12 +1,26 @@
-import { File as FileType } from '@unchainedshop/core-files';
+import { File as FileType, getFileAdapter } from '@unchainedshop/core-files';
 import { Context } from '../../context.js';
-
+import { checkAction } from '../../acl.js';
+import { actions } from '../../roles/index.js';
 export interface MediaHelperTypes {
-  url: (language: FileType, params: Record<string, any>, context: Context) => string;
+  url: (language: FileType, params: Record<string, any>, context: Context) => Promise<string>;
 }
 
 export const Media: MediaHelperTypes = {
-  url(root, params, { modules }) {
-    return modules.files.getUrl(root, params);
+  url: async (file, params, context) => {
+    const { modules } = context;
+    try {
+      await checkAction(context, actions.downloadFile, [file, params]);
+      const fileUploadAdapter = getFileAdapter();
+      const mediaUrl = modules.files.getUrl(file, params);
```

Comment:
Url Transformation should come after getting the original URL and signing from the provider. getUrl should be able to transform the URL to CDN or Thumbor.